firebase-project/src/views/register.vue

```diff
@@ -0,0 +1,252 @@
+        <div class="fixLabel1">
+          <label>First name</label>
+          <Field
+            v-model="firstName"
+            name="firstName"
+            type="firstName"
+            class="form-control"
+            placeholder="First name"
+            :rules="validatefirstName"
+          />
+          <ErrorMessage class="validateErr" name="firstName" />
+        </div>
+        <div class="fixLabel2">
+          <label>Last name</label>
+          <Field
+            v-model="lastName"
+            name="lastName"
+            type="lastName"
+            class="form-control"
+            placeholder="Last name"
+            :rules="validatelastName"
+          />
+          <ErrorMessage class="validateErr" name="lastName" />
+        </div>
+      </div>
+
+      <label>Mobile Number</label>
+      <Field
+        v-model="mobileNumber"
+        name="mobileNumber"
+        type="number"
+        class="form-control"
+        placeholder="Mobile Number"
+        :rules="validatemobileNumber"
+      />
+      <ErrorMessage class="validateErr" name="mobileNumber" />
+      <label>Profile Photo</label>
+      <Field
+        v-model="profilePhoto"
+        name="profilePhoto"
+        type="file"
+        class="form-control"
+        :rules="validateprofilePhoto"
+      />
+      <ErrorMessage class="validateErr" name="profilePhoto" />
+      <label>Email</label>
+      <Field
+        v-model="email"
+        name="email"
+        type="email"
+        class="form-control"
+        placeholder="Email"
+        :rules="validateEmail"
+      />
+      <ErrorMessage class="validateErr" name="email" />
+      <label>Password</label>
+      <Field
+        v-model="password"
+        name="password"
+        type="password"
+        class="form-control"
+        placeholder="Password"
+        :rules="validatePassword"
+      />
+      <ErrorMessage class="validateErr" name="password" />
+
+      <label>Confirm Password</label>
+      <Field
+        v-model="confirmPassword"
+        name="confirmPassword"
+        type="password"
+        class="form-control"
+        placeholder="Confirm Password"
+        :rules="validateConfirmPassword"
+      />
+      <ErrorMessage class="validateErr" name="confirmPassword" />
+      <div>
+        <button type="submit" class="submitBtn">REGISTER NOW</button>
+      </div>
+      <div v-if="error">{{ error }}</div>
+    </Form>
+  </div>
+</template>
+
+<script>
+import { Form, Field, ErrorMessage } from "vee-validate";
+import { ref } from "vue";
+import { useStore } from "vuex";
+import { useRouter } from "vue-router";
+
+export default {
+  name: "register-page",
+  components: {
+    Form,
+    Field,
+    ErrorMessage,
+  },
+  setup() {
+    const email = ref("");
+    const password = ref("");
+    const firstName = ref("");
+    const lastName = ref("");
+    const mobileNumber = ref("");
+    const profilePhoto = ref("");
+    const confirmPassword = ref("");
```

Comment:
can you remove all this ref for from and create a reactive variable form = reactive({
email: '',
password: ''.........so on
})
@mukeshsin